docs/VARIABLES.md

```diff
@@ -2380,6 +2380,86 @@ Default:  "{{mds_ssl_enabled}}"
 
 ***
 
+### sso_mode
+
+SSO mode for C3. Possible values: oidc. if enabling oidc you must set  sso_groups, sso_sub, sso_jwks_uri, sso_authorize_uri, sso_token_uri, sso_issuer_url, sso_client_id, sso_client_password in MDS
+
+Default:  none
+
+***
+
+### sso_groups
+
+Groups in JWT
+
+Default:  none
+
+***
+
+### sso_sub
+
+Sub in JWT
+
+Default:  none
+
+***
+
+### sso_issuer_url
+
+The issuer url, which is typically the authorization server's URL
+
+Default:  none
+
+***
+
+### sso_jwks_uri
+
+JSON Web Key Set (JWKS) URI
+
+Default:  none
+
+***
+
+### sso_authorize_uri
+
+Base uri for authorize endpoint, that initiates an OAuth authorization request
+
+Default:  none
+
+***
+
+### sso_token_uri
+
+Base uri for IdP token endpoint, from where a token is requested by MDS
+
+Default:  none
+
+***
+
+### sso_client_id
+
+Client id for authorize and token request to Idp
+
+Default:  none
+
+***
+
+### sso_client_password
+
+Client password for authorize and token request to Idp
+
+Default:  none
+
+***
+
+### sso_scope
+
+If any additional scope is needed to include groups in the token, this config is optional based on Idp
+
+Default:  none
+
+***
+
```

Comment:
There are other configs as well which have default values but can be configured.

> confluent.oidc.idp.refresh.token.enabled=true

> confluent.oidc.session.token.expiry.ms=900000
> confluent.oidc.session.max.timeout.ms=21600000

We should add variable for above configs as well.

Reply:
we usually don't add variables for timeouts, does it add any value to make the timeout configurable for the customer? for confluent.oidc.idp.refresh.token.enabled=true it would make sense to add a variable

Reply:
> does it add any value to make the timeout configurable for the customer

I agree - most of customers would not but I think few customer may want to configure that. 
If it can be easily added in future, upon a customer's request, you can skip this for now.

Reply:
yes, we provide a functinality `kafka_broker_custom_properties` using that, the customer can override any config. The can use it to modify timeout in this case. For refresh token, I think it would add more value to have a variable. Does this sound good?

Reply:
Yes, this seems good.